.circleci/config.yml

```
@@ -19,21 +19,51 @@ commands:
   check-uncommitted:
     description: Check for uncommitted changes.
     steps:
-      - run: 
-          name: Check for uncomitted changes using git porcelain command.
+      - run:
+          name: Check for uncomitted changes.
           command: |
             if [[ `git status --porcelain` ]]; then
               echo "ERROR: uncommitted changes"
               git status --porcelain
               exit 1
             fi
+  install-golangci-lint:
+    description: Install golangci-lint
+    parameters:
+      version:
+        type: string
+        default: 1.21.0
+      gobin:
+        type: string
+        default: /go/bin
+      prefix:
+        type: string
+        default: v1
+        description: Prefix for cache key to store the binary.
+    steps:
+      - restore_cache:
+          name: Restore golangci-lint cache
+          keys: ['<< parameters.prefix >>-golangci-lint-{{ arch }}-<< parameters.version >>']
+      - run:
+          name: Install golangci-lint
+          command: |
+            mkdir -p << parameters.gobin >>
+            command -v << parameters.gobin >>/golangci-lint && exit
+
+            download=https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh
+            wget -O- -q $download | sh -s -- -b << parameters.gobin >>/ v<< parameters.version >>
```

I would vote for not running this completely silenced.
```suggestion
            wget -nv- -q $download | sh -s -- -b << parameters.gobin >>/ v<< parameters.version >>
```

---

Looks like the Mattermost server already silences the `curl` -- any specific concern here?

---

No specific ones. I seams like one of the cases where if something does wrong, it want to see the error message. 1/5